### PR TITLE
feat(op): timestamp below bedrock

### PR DESCRIPTION
**Pull request by emhane**
Allows any block below bedrock to have same timestamp as its parent

### CODE REVIEW

**Comment by joshieDo** [comment]
Headers are just a placeholders below bedrock right? Instead of this, can we just fake the timestamp to be a counter?

**Comment by emhane** [comment]
> Headers are just placeholders below bedrock right? Instead of this, can we just fake the timestamp to be a counter?

no they are real headers, we are importing them. however, this check isn't lax enough. have come across blocks where the child is older than the parent by almost 10 minutes. thinking of just disabling check for below bedrock, i.e. below op-merge, blocks. wdyt?
block 900066: `864 days 22 hrs ago (Dec-07-2021 07:29:11 PM +UTC)`
block 900067: `864 days 22 hrs ago (Dec-07-2021 07:20:42 PM +UTC)`
(src: https://otterscan.mainnet.testinprod.io)

**Comment by emhane** [comment]
re-request @joshieDo

File: crates/primitives/src/header.rs [hunk 1]
```diff
@@ -775,7 +775,17 @@ impl SealedHeader {
         }
 
         // timestamp in past check
-        if self.header.is_timestamp_in_past(parent.timestamp) {
+        if chain_spec.is_optimism_mainnet() {
+            #[cfg(feature = "optimism")]
```
Comment (joshieDo) [inline crates/primitives/src/header.rs:779]:
```rust
if cfg!(feature = "optimism") && chain_spec.is_optimism_mainnet()  &&
...
```

Comment (emhane) [reply crates/primitives/src/header.rs:779]: `is_bedrock_active_at_block` is op feature gated

Comment (joshieDo) [reply crates/primitives/src/header.rs:779]:
```rust
if cfg!(feature = "optimism") {
   #[cfg(feature = "optimism")] 
   if ... {
   }
}
```

maybe too much of a nit, but i just didnt want that check to happen every time, and let it be optimized at compilation

**Review by joshieDo — APPROVED** [review]
lgtm, apart from that nit

**Review by mattsse — APPROVED** [review]

**Comment by emhane** [comment]
@mattsse, for failing integration test `node-e2e-tests::p2p::can_sync`, it's either adding op feature to this test crate in order to check for block below bedrock, because it requires importing `reth-primitves/optimism` into `dev-dependencies` which will otherwise conflict. or excluding this test on runs with optimism feature in ci completely. wdyt?

**Comment by joshieDo** [comment]
i'd rather not disable the test, i'll take alook into it since it was a newly test added recently

File: crates/primitives/src/header.rs [hunk 2]
```diff
@@ -776,6 +776,18 @@ impl SealedHeader {
         }
 
         // timestamp in past check
+        #[cfg(feature = "optimism")]
+        if !chain_spec.is_optimism_mainnet() ||
```
Comment (joshieDo) [inline crates/primitives/src/header.rs:780]: this || is wrong here, otherwise every OP chain will fail the header validation

**Pull request merged**